client.go

```diff
@@ -936,8 +937,8 @@ func clientDoDeadline(req *Request, resp *Response, deadline time.Time, c client
 	case err = <-ch:
 		if resp != nil {
 			respCopy.copyToSkipBody(resp)
-			swapResponseBody(resp, respCopy)
 		}
+		swapResponseBody(resp, respCopy)
```

Comment:
Are you sure you want to move this out of the if check? Calling `swapResponseBody` with `resp == nil` will cause a panic.